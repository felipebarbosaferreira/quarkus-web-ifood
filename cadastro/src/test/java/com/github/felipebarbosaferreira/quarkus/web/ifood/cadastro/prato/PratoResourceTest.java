package com.github.felipebarbosaferreira.quarkus.web.ifood.cadastro.prato;

import static io.restassured.RestAssured.given;
import static org.hamcrest.Matchers.equalTo;

import java.math.BigDecimal;

import javax.ws.rs.core.MediaType;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.felipebarbosaferreira.quarkus.web.ifood.cadastro.PostgresDBLifeCycleControlTest;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(PostgresDBLifeCycleControlTest.class)
public class PratoResourceTest {

	@Test
	@DataSet("test-prato-listar-todos-1.yml")
	public void testListarTodosEndpoint() {
		String retorno = given().when().get("/pratos").then().statusCode(200).extract().asString();
		Approvals.verifyJson(retorno);
	}

	@Test
	@DataSet("test-prato-listar-todos-1.yml")
	public void testListarTodosPorRestauranteEndpoint() {
		// TODO - yml com objetos restaurantes
		String retorno = given().when().get("/pratos/restaurante/8").then().statusCode(200).extract().asString();
		Approvals.verifyJson(retorno);
	}

	@Test
	@DataSet("test-prato-listar-todos-1.yml")
	public void testObterEndpoint() {
		String retorno = given().when().get("/pratos/7").then().statusCode(200).extract().asString();
		Approvals.verifyJson(retorno);
	}

	@Test
	public void testAdicionarEndpoint() {
		PratoDomain pratoDomainMock = getMockSavePratoTest();
		given().body(pratoDomainMock).contentType(MediaType.APPLICATION_JSON).when().post("/pratos").then()
				.statusCode(204);
		given().when().get("/pratos/1").then().statusCode(200).body("nome", equalTo(pratoDomainMock.nome));
	}

	@Test
	@DataSet("test-prato-listar-todos-1.yml")
	public void testAtualizarEndpoint() {
		PratoDomain pratoDomainMock = getMockSavePratoTest();
		given().body(pratoDomainMock).contentType(MediaType.APPLICATION_JSON).when().put("/pratos/7").then()
				.statusCode(204);
		given().when().get("/pratos/7").then().statusCode(200).body("nome", equalTo(pratoDomainMock.nome));
	}

	@Test
	public void testDeletarEndpoint() {
		given().contentType(MediaType.APPLICATION_JSON).when().delete("/pratos/8").then().statusCode(204);
		given().when().get("/pratos/8").then().statusCode(404);
	}

	private PratoDomain getMockSavePratoTest() {
		return new PratoDomain("Peixada Cearense", 
				"Filé de peixe, tomate, cebola, pimentões, leite de coco, batatas, cenouras, repolho e ovos cozidos. Acompanha pirão e arroz branco.", 
				1L, 
				new BigDecimal(24.12));
	}

}