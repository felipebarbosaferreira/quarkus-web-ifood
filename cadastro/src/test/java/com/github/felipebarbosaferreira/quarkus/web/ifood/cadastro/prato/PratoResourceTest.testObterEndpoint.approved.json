{
  "descricao": "Receita contemporânea. Sabor agridoce. Filé de peixe grelhado com pedacinhos de castanha de caju e molho de manga com gengibre e maracujá. Acompanha arroz com brócolis e purê de abóbora.",
  "id": 7,
  "nome": "Peixe Praiano",
  "preco": 32.12
}